plugins/package-managers/swiftpm/src/main/kotlin/SwiftPm.kt

```diff
@@ -159,7 +172,21 @@ class SwiftPm(
     }
 }
 
-private fun parseLockfile(packageResolvedFile: File): Set<Package> {
-    val resolved = packageResolvedFile.inputStream().use { json.decodeFromStream<PackageResolved>(it) }
-    return resolved.objects["pins"].orEmpty().mapTo(mutableSetOf()) { it.toPackage() }
+private fun parseLockfile(packageResolvedFile: File): Result<Set<Package>> {
+    val root: JsonObject = packageResolvedFile.inputStream().use { json.decodeFromStream(it) }
+    val version = root.getValue("version").jsonPrimitive.content
+
+    return when {
+        version == "1" -> {
+            val pinsJson = root["object"]?.jsonObject?.get("pins")?.toString() ?: "{}"
+            val pins = json.decodeFromString<List<PinV1>>(pinsJson)
```

Comment:
There's no need to convert back to a string and forth to a type again, just use `decodeFromJsonElement()`. I'll add this to my code proposal above.